astropy/utils/iers/iers.py

```diff
@@ -443,5 +520,143 @@ def pm_source(self, i):
+
+        return cls.iers_table
+
+    def _check_interpolate_indices(self, indices_orig, indices_clipped, max_input_mjd):
+        """Check that the indices from interpolation match those after clipping to the
+        valid table range.  The IERS_Auto class is exempted as long as it has
+        sufficiently recent available data so the clipped interpolation is
+        always within the confidence bounds of current Earth rotation
+        knowledge.
+        """
+        predictive_mjd = self.meta['predictive_mjd']
+
+        # See explanation in _refresh_table_as_needed for these conditions
+        auto_max_age = conf.auto_max_age if (conf.auto_max_age is not None) else 1e30
+        if (max_input_mjd > predictive_mjd and
+                self.time_now.mjd - predictive_mjd > auto_max_age):
+            raise ValueError(INTERPOLATE_ERROR)
+
+    def _refresh_table_as_needed(self, mjd):
+        """Potentially update the IERS table in place depending on the requested
+        time values in ``mjd`` and the time span of the table.
+
+        For IERS_Auto the behavior is that the table is refreshed from the IERS
+        server if both the following apply:
+
+        - Any of the requested IERS values are predictive.  The IERS-A table
+          contains predictive data out for a year after the available
+          definitive values.
+        - The first predictive values are at least ``conf.auto_max_age days`` old.
+          In other words the IERS-A table was created by IERS long enough
+          ago that it can be considered stale for predictions.
+        """
+        max_input_mjd = np.max(mjd)
+        now_mjd = self.time_now.mjd
+
+        # IERS-A table contains predictive data out for a year after
+        # the available definitive values.
+        fpi = self.meta['predictive_index']
+        predictive_mjd = self.meta['predictive_mjd']
+
+
+        # Update table in place if necessary
+        auto_max_age = conf.auto_max_age if (conf.auto_max_age is not None) else 1e30
+
+        # If auto_max_age is smaller than IERS update time then repeated downloads may
+        # occur without getting updated values (giving a IERSStaleWarning).
+        if auto_max_age < 10:
+            raise ValueError('IERS auto_max_age configuration value must be larger than 10 days')
+
+        if (max_input_mjd > predictive_mjd and
+               now_mjd - predictive_mjd > auto_max_age):
+
+            # Get the latest version
+            try:
+                clear_download_cache(conf.iers_auto_url)
+                filename = download_file(conf.iers_auto_url, cache=True)
+            except Exception as err:
+                # Issue a warning here, perhaps user is offline.  An exception
+                # will be raised downstream when actually trying to interpolate
+                # predictive values.
+                warn(AstropyWarning('failed to download {}: {}'
+                                    .format(conf.iers_auto_url, str(err))))
```

Comment:
I think this warning needs to be more verbose, along the lines of the old (removed in this PR) `_IERS_HINT`.  It should specify why the user might care (might compromise precision or lead to later errors) and also should tell the user how to try again if they re-connect their internet.  E.g.,
```
If you re-establish your internet and want to update to the latest IERS in this python session, you will need to do:
     >>> from astropy.utils import iers
     >>> iers.IERS_Auto.iers_table = iers.IERS_Auto.open()
```